lib/Service/FileService.php

```diff
@@ -88,7 +90,14 @@ public function getShareLink(IShare $share): string
 	 */
 	private function getCurrentDomain(): string
 	{
-		return $this->urlGenerator->getBaseUrl();
+		$baseUrl = $this->urlGenerator->getBaseUrl();
+		$trustedDomains = $this->config->getSystemValue('trusted_domains');
+
+		if(isset($trustedDomains[1]) === true) {
```

Comment:
Whats in trustedDomains[0] ?

Reply:
trustedDomains[0] is almost always localhost, and [1] then contains the application domain.